src/lib/DepositRecordSerializer.js

```diff
@@ -41,8 +45,69 @@ export class DepositRecordSerializer {
     return _isNumber(obj) || obj ? obj : null;
   }
 
+  /**
+  * Transform frontend creators structure to API-compatible structure.
+  *
+  * NOTE: Serialization doesn't deal with validation: safely access properties
+  * NOTE: If property absent from input, it should be absent from output
+  * @method
+  * @param {object} record - with creators in frontend format
+  * @returns {object} record - with creators in API format
+  */
+  serializeCreators(record) {
+    const in_creators = record.creators || [];
```

Comment:
@fenekku why should we change the `identifiers` from an object in the backend to an array in the frontend? I think is better to be consistent with the fields' schema. Is there any particular reason that I miss?

Reply:
We do this dance of backend object -> frontend array -> to backend object because the component used to fill the identifiers is an array component. The user can add many different identifiers and remove them: visually it's an array. We don't have a component that is visually like an array but backs to an object, so the ArrayField was used and it does the job :smile:

Reply:
After our discussion, I created a generic issue for this matter https://github.com/inveniosoftware/react-invenio-deposit/issues/42

Reply:
:+1: